src/AstArgument.h

```diff
@@ -41,11 +41,13 @@ namespace souffle {
  */
 class AstVariable : public AstArgument {
 public:
-    AstVariable(std::string name) : name(std::move(name)) {}
+    AstVariable(std::string name, SrcLocation loc = {}) : name(std::move(name)) {
+        setSrcLoc(std::move(loc));
+    }
```

Comment:
I think these are an improvement, but would it perhaps be even better to push location up to the AstNode constructor?
```suggestion
    AstVariable(std::string name, SrcLocation loc = {}) : AstArgument(loc), name(std::move(name)) { }
```